src/mcts/search.cc

```diff
@@ -1524,7 +1507,96 @@ void SearchWorker::DoBackupUpdateSingleNode(
   search_->total_playouts_ += node_to_process.multivisit;
   search_->cum_depth_ += node_to_process.depth * node_to_process.multivisit;
   search_->max_depth_ = std::max(search_->max_depth_, node_to_process.depth);
-}  // namespace lczero
+}
+
+bool SearchWorker::MaybeRestrictOutcome(Node* p, float m) const {
+  auto found_tb = false;
+  auto losing_m = 0.0f;
+
+  // Determine the maximum (Lower, Upper) bounds across all children.
+  // (-1,-1) Loss (initial and lowest bounds)
+  // (-1, 0) Can't Win
+  // (-1, 1) Regular node
+  // ( 0, 0) Draw
+  // ( 0, 1) Can't Lose
+  // ( 1, 1) Win (highest bounds)
+  auto max_L = -1;
+  auto max_U = -1;
+  for (const auto& edge : p->Edges()) {
+    switch (edge.GetNodeType()) {
+      case Node::NodeType::Terminal:
+      case Node::NodeType::TablebaseTerminal: {
+        found_tb = found_tb || edge.IsTbTerminal();
+
+        // Default_wl doesn't matter as WL is only used if IsTerminal is true.
+        const auto WL = edge.GetWL(0.0f);
+        // We can short-circuit after a single win maxes the bounds.
+        if (WL > 0.0f) {
+          max_L = max_U = 1;
+          break;
```

Comment:
Oops, this short-circuit is broken with the `switch` but looks like we'll need to iterate over all wins anyway to find the shortest (non-tb) win anyway.

Reply:
Removed comment and break.